add options for colouring conservation scores

sillitoe opened the pull request:
Provides some more flexibility for specifying how conservation scores are displayed (including changing the colour according to the height of each bar).

See ```snippets/conserv``` for an example.

adds the following options to the main config:

```
conserv: {
	maxHeight: 20,
	fillColor: ['#660', '#ff0'],
	strokeColor: '#000'
}
```

```maxHeight``` is the height of the component (pixels) 

```fillColor``` / ```strokeColor```:

 - If a single value is given then all the elements are coloured the same
 - If a [min,max] range is given then each element is coloured in a sliding scale according to
the height.

Colours can be specified as anything that D3 understands e.g. "#00F", "rgb(0,0,1)", "blue" (see https://github.com/d3/d3-color)

NOTE: this adds ```d3-scale``` as a dependency which seems to add ~25K to the min.gz. I would argue that this is worth it as being able to interpolate (and translate) colours is pretty useful generally.

wilzbach commented:
> seems to add ~25K to the min.gz.

Ouch this is 50% of the file.. Maybe we can make this optional? I will have a look at your code at the weekend ;-)

sillitoe commented:
Yup, that was annoying. I've just been looking at trying to bring in a limited subset of d3-scale with reduced functionality and fewer dependencies, but it's going to take me too long (and it seems a pretty bad idea philosophically).

I think it's a good idea to make this an optional dependency (though I'm not sure how to code that). Essentially d3 is used for two things here:

1. provide a sliding color scale
1. provide flexibility when defining colors

Both points make for a nice user interface, but are optional.

1. only required if a range is specified, otherwise just use whatever fixed string is provided for all elements (this is still useful)
1. make the user specify a colour in a preset format

An alternative would be to allow the user to specify their own function to munge the data structure given to the ```svg.rect()```.

```
d = { x: x, y: y, height: height, ... };
d = this.userDefinedDataMunger( d );
svg.rect(d);
```

Would allow the same functionality, but would leave it to the user to implement (I'll write that up as a separate PR).

sillitoe commented:
Okay, I've removed the d3 dependency in the package.json and it does something sensible if d3_scale isn't found (warns and accept first color in range).

TODO: ```require("d3-scale")``` will fail if this hasn't been installed.

I've also added an optional function to explicitly style each item (sorry, I should've done this in separate commits)

sillitoe commented:
Tidied this up in #130
